### PR TITLE
[JUJU-3209] Sync CLI docs to Discourse

### PR DESCRIPTION
**Pull request by barrettj12**
Add a new script `scripts/discourse-sync` to sync the output of the `juju documentation` command to Discourse. Invoke the script like so:
```bash
set -eux
export DISCOURSE_API_USERNAME=[your-discourse-username]
export DISCOURSE_API_KEY=[api-key] # ask Dora for one
export DOCS_DIR=./docs # or $(mktemp -d)
export TOPIC_IDS=./.github/discourse-topic-ids.yaml

juju documentation --split --out=$DOCS_DIR --discourse-ids $TOPIC_IDS # --no-index
python3 ./scripts/discourse-sync/main.py
```

Also add a new GitHub Action which uses this script to sync our docs to Discourse.

### Notes
- Tested with Python 3.8.10 locally / whatever version is pre-installed on the `ubuntu-latest` runners.
- The script will get the current post from the Discourse API, and check for changes before deciding to push a new version.
- You can see the results [here](https://discourse.charmhub.io/t/juju-cli-commands/10045).
- The autogenerated docs are not yet visible on [juju.is/docs](https://juju.is/docs) - there is a bit more work to do on the output of `juju documentation`. Once we are happy with it, we just need to update the topic IDs in `discourse-topic-ids.yaml`.
- See [here](https://github.com/barrettj12/juju/actions/runs/4761021644/jobs/8461886772) for a sample workflow run (triggered manually with `gh workflow run docs.yml -r doc-sync`).
- This workflow will run on `push` (new updates) to the `develop` branch. Long term, we want this to run on the LTS branch, but `develop` is currently the only branch which has the updated `juju documentation` command.

### CODE REVIEW

**Comment by juanmanuel-tirado** [comment]
The script invocation probably need to be something like `juju documentation --split --out=[docs-dir] --url=https://discourse.charmhub.io/docs/idontknowhat/t` to generate the `see-also` url correctly. Then this should generate something like 
```md
**See also:** 

- [`credentials`](https://discourse.charmhub.io/docs/idontknowhat/t/credentials)
- [`remove-credential`](https://discourse.charmhub.io/docs/idontknowhat/t/remove-credentials)
```

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: "Docs"
+on:
+  push:
+    branches:
+      - 'develop'
```
Comment (SimonRichardson) [inline .github/workflows/docs.yml:5]: I wonder if this should actually be the LTS or the one behind develop. As develop things are new and very much edge edge.

Comment (barrettj12) [reply .github/workflows/docs.yml:5]: Fair point. At the moment it needs to be develop, as this is the only branch which has the `juju documentation` command. But long term, it makes sense to use the latest LTS branch. (Or even version the docs by branch, but that's another discussion).

**Comment by juanmanuel-tirado** [comment]
After #15483 lands, the `documentation` command can be changed to `juju documentation --split --out=$DOCS_DIR --no-index --discourse-ids $POST_IDS` to use the discourse ids.

**Review by juanmanuel-tirado — APPROVED** [review]
R2GO after changing the `documentation` command params as mentioned above.

**Pull request merged**